src/Stratis.Bitcoin.Features.PoA/PoANetwork.cs

```diff
@@ -0,0 +1,179 @@
+                powAllowMinDifficultyBlocks: false,
+                powNoRetargeting: false,
+                powLimit: null,
+                minimumChainWork: null,
+                isProofOfStake: false,
+                lastPowBlock: 0,
+                proofOfStakeLimit: null,
+                proofOfStakeLimitV2: null,
+                proofOfStakeReward: Money.Zero
+            );
+
+            // https://en.bitcoin.it/wiki/List_of_address_prefixes
+            this.Base58Prefixes = new byte[12][];
+            this.Base58Prefixes[(int)Base58Type.PUBKEY_ADDRESS] = new byte[] { (55) }; // 'P' prefix
+            this.Base58Prefixes[(int)Base58Type.SCRIPT_ADDRESS] = new byte[] { (117) }; // 'p' prefix
+            this.Base58Prefixes[(int)Base58Type.SECRET_KEY] = new byte[] { (63 + 128) };
+            this.Base58Prefixes[(int)Base58Type.ENCRYPTED_SECRET_KEY_NO_EC] = new byte[] { 0x01, 0x42 };
+            this.Base58Prefixes[(int)Base58Type.ENCRYPTED_SECRET_KEY_EC] = new byte[] { 0x01, 0x43 };
+            this.Base58Prefixes[(int)Base58Type.EXT_PUBLIC_KEY] = new byte[] { (0x04), (0x88), (0xB2), (0x1E) };
+            this.Base58Prefixes[(int)Base58Type.EXT_SECRET_KEY] = new byte[] { (0x04), (0x88), (0xAD), (0xE4) };
+            this.Base58Prefixes[(int)Base58Type.PASSPHRASE_CODE] = new byte[] { 0x2C, 0xE9, 0xB3, 0xE1, 0xFF, 0x39, 0xE2 };
+            this.Base58Prefixes[(int)Base58Type.CONFIRMATION_CODE] = new byte[] { 0x64, 0x3B, 0xF6, 0xA8, 0x9A };
+            this.Base58Prefixes[(int)Base58Type.STEALTH_ADDRESS] = new byte[] { 0x2a };
+            this.Base58Prefixes[(int)Base58Type.ASSET_ID] = new byte[] { 23 };
+            this.Base58Prefixes[(int)Base58Type.COLORED_ADDRESS] = new byte[] { 0x13 };
+
+            this.Checkpoints = new Dictionary<int, CheckpointInfo>
+            {
+                //TODO POA add checkpoints
+                { 0, new CheckpointInfo(new uint256("0x30edbee76ea7338cbd039a1a6fa41e94256f81994ab12b4f3aa9e01104808c38")) },
+            };
+
+            var encoder = new Bech32Encoder("bc");
+            this.Bech32Encoders = new Bech32Encoder[2];
+            this.Bech32Encoders[(int)Bech32Type.WITNESS_PUBKEY_ADDRESS] = encoder;
+            this.Bech32Encoders[(int)Bech32Type.WITNESS_SCRIPT_ADDRESS] = encoder;
+
+            this.DNSSeeds = new List<DNSSeedData>
+            {
+                // TODO POA
+                //new DNSSeedData("seednode1.stratisplatform.com", "seednode1.stratisplatform.com"),
+                //new DNSSeedData("seednode2.stratis.cloud", "seednode2.stratis.cloud"),
+                //new DNSSeedData("seednode3.stratisplatform.com", "seednode3.stratisplatform.com"),
+                //new DNSSeedData("seednode4.stratis.cloud", "seednode4.stratis.cloud")
+            };
+
+            // TODO POA change IPs here
+            string[] seedNodes = { "101.200.198.244" };
+            this.SeedNodes = this.ConvertToNetworkAddresses(seedNodes, this.DefaultPort).ToList();
```

Comment:
a fluent method in this class, like
`public PoANetwork WithPoANetworkSettings(Action<PoANetworkSettings>)`
that expose a class PoANetworkSettings with the needed parameters to setup the network could be useful?

this way in the Program.cs of PoAChainD you can use it as
```
Network network = new PoANetwork().WithPoANetworkSettings(settings =>{
   settings.SeedNodes = ....
};
```

this way you are not forced to mess with PoANetwork for setup your federated configuration

Reply:
For that we can do

```
Network network = new PoANetwork()
                {
                    SeedNodes = { ... }
                };
```
in daemon, so probably not needed for now but not a bad idea 🙂

Reply:
the difference is that with PoANetworkSettings you can have only settings for a PoA network that differs from default Network. Not sure if it's needed tho